### ISSUE TITLE
Ensure ECE Implementation Supports the Same Plugins and Product Types as PRBs

### ISSUE THREAD

**Issue by rafaelzaleski** [Automattic/woocommerce-payments#8871]
Review the current implementation of the Stripe Express Checkout Element (ECE) api and ensure we are supporting the same product types and plugins as PRBs. 

PRBs are currently declaring support for the following product types (some are created by external plugins):

````
'simple',
'variable',
'variation',
'subscription',
'variable-subscription',
'subscription_variation',
'booking',
'bundle',
'composite',
'mix-and-match',
````

### Acceptance criteria:

- The ECE implementation supports the same product types/plugins as PRBs.
- Ensure there are no regressions in Payment Request Buttons (PRB) when ECE is disabled.
- There are no regressions when no express checkout payment methods are active.

**Comment by bborman22** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Fractal-62a0f7b955648b001e1f5e1f/issues/Automattic/woocommerce-payments/8871?planning-poker) with Zenhub @asumaran @cesarcosta99 @lovo-h @rafaelzaleski @reykjalin @ricardo

**Comment by rafaelzaleski** [comment]
@asumaran, I noticed that although we haven’t officially declared support for product types related to **WooCommerce deposits**, there is code related to this in the PRBs JS files. **Could you verify whether PRBs support WooCommerce Deposits and if this functionality is mirrored in ECE?** 
If not, could you look into this? (in a new PR after #8987 please 😅 )

**Comment by asumaran** [comment]
@rafaelzaleski I'll handle it here https://github.com/Automattic/woocommerce-payments/issues/9067
